src/Sylius/Bundle/ApiBundle/Serializer/ProductVariantNormalizer.php

```diff
@@ -23,23 +23,23 @@
 use Sylius\Component\Core\Model\ChannelInterface;
 use Sylius\Component\Core\Model\ProductVariantInterface;
 use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
-use Symfony\Component\Serializer\Normalizer\ContextAwareNormalizerInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerAwareInterface;
 use Symfony\Component\Serializer\Normalizer\NormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\NormalizerInterface;
 use Webmozart\Assert\Assert;
 
 /** @experimental */
-final class ProductVariantNormalizer implements ContextAwareNormalizerInterface, NormalizerAwareInterface
+final class ProductVariantNormalizer implements NormalizerInterface, NormalizerAwareInterface
 {
     use NormalizerAwareTrait;
 
     private const ALREADY_CALLED = 'sylius_product_variant_normalizer_already_called';
 
     public function __construct(
-        private ProductVariantPricesCalculatorInterface $priceCalculator,
-        private AvailabilityCheckerInterface $availabilityChecker,
-        private SectionProviderInterface $uriBasedSectionContext,
-        private IriConverterInterface $iriConverter,
+        private readonly ProductVariantPricesCalculatorInterface $priceCalculator,
```

Comment:
`NormalizerAwareTrait` blocks making class readonly?

Might as well drop `NormalizerAwareTrait` usage and inject normalizer directly.
![image](https://github.com/Sylius/Sylius/assets/870747/babccb20-3bfa-4aaa-9684-a31481f512b5)

This trait feel legacy-like is not related to `NormalizerInterface` in any way, while sounding too similar

I vote for making a policy out of this case and inject normalizer directly in every `ApiBundle/Serializer`.

WDYT?

Reply:
Yes, `NormalizerAwareTrait` blocks making class readonly. And I tried to do what you suggested, but I got a circular reference:  `"serializer -> Sylius\Bundle\ApiBundle\Serializer\ProductVariantNormalizer -> serializer"`.